### PR TITLE
Add diff tab to code size page

### PR DESCRIPTION
**Pull request by peterdjlee**
* Handle negative byte sizes in both treemap and tree table
Screen Shot 2020-07-01 at 3 38 58 PM

### CODE REVIEW

File: packages/devtools_app/lib/src/charts/treemap.dart [hunk 1]
```diff
@@ -84,7 +84,7 @@ class Treemap extends StatelessWidget {
     endIndex ??= nodes.length - 1;
     int sum = 0;
     for (int i = startIndex; i <= endIndex; i++) {
-      sum += nodes[i].byteSize ?? 0;
+      sum += nodes[i].absByteSize ?? 0;
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/charts/treemap.dart:87]: perhaps rename to `unsignedByteSize`

File: packages/devtools_app/lib/src/charts/treemap.dart [hunk 2]
```diff
@@ -517,6 +518,19 @@ class TreemapNode extends TreeNode<TreemapNode> {
   final Map<String, TreemapNode> childrenMap;
   int byteSize;
 
+  final bool showDiff;
+
+  int get absByteSize => byteSize.abs();
+  
+
+  Color displayColor() {
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/charts/treemap.dart:526]: make this a getter

File: packages/devtools_app/lib/src/code_size/code_size_controller.dart [hunk 3]
```diff
@@ -10,6 +10,7 @@ import 'package:path/path.dart';
 import 'package:vm_snapshot_analysis/treemap.dart';
 
 import '../charts/treemap.dart';
+import 'stub_data/fake_data.dart';
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/code_size/code_size_controller.dart:13]: let's rename this file to fake_diff_data

File: packages/devtools_app/lib/src/code_size/code_size_screen.dart [hunk 4]
```diff
@@ -62,30 +85,69 @@ class CodeSizeBodyState extends State<CodeSizeBody> with AutoDisposeMixin {
...
+      _buildTreemapAndTableSplitView(showDiff: false),
+      _buildTreemapAndTableSplitView(showDiff: true),
+    ];
+  }
+
+  Widget _buildTreemapAndTableSplitView({@required bool showDiff}) {
+    return Split(
+      axis: Axis.vertical,
+      children: [
+        _buildTreemap(),
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/code_size/code_size_screen.dart:141]: Add a TODO that says we should try to reuse the same treemap widget and only swap out the tables in Diff mode

File: packages/devtools_app/lib/src/code_size/code_size_table.dart [hunk 5]
```diff
@@ -125,3 +125,69 @@ class _SizePercentageColumn extends ColumnData<TreemapNode> {
   @override
   double get fixedWidthPx => 100.0;
 }
+
+class CodeSizeDiffTable extends StatelessWidget {
+  factory CodeSizeDiffTable({@required rootNode}) {
+    final treeColumn = _NameColumn(currentRootLevel: rootNode.level);
+    final columns = List<ColumnData<TreemapNode>>.unmodifiable([
+      treeColumn,
+      _DiffColumn(),
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/code_size/code_size_table.dart:134]: can we include the diff bytes and percent columns as well? like here: https://pub.dev/packages/vm_snapshot_analysis#compare. you may need to look into the vm_snapshot_analysis package code to see how percent was calculated

File: packages/devtools_app/lib/src/code_size/code_size_table.dart [hunk 6]
```diff
@@ -125,3 +125,69 @@ class _SizePercentageColumn extends ColumnData<TreemapNode> {
...
+      dataRoots: rootNode.children,
+      columns: columns,
+      treeColumn: treeColumn,
+      keyFactory: (node) => PageStorageKey<String>(node.name),
+      sortColumn: treeColumn,
+      sortDirection: SortDirection.ascending,
+    );
+  }
+}
+
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/code_size/code_size_table.dart:167]: Add a TODO to add an opaque overlay / background like in the mocks

File: packages/devtools_app/lib/src/table.dart [hunk 7]
```diff
@@ -1124,6 +1124,12 @@ class _TableRowState<T> extends State<TableRow<T>>
     );
   }
 
+  TextStyle columnTextStyle(ColumnData<T> column) {
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/table.dart:1127]: rename to contentTextStyle as it is not the style for the whole column

File: packages/devtools_app/lib/src/table_data.dart [hunk 8]
```diff
@@ -393,6 +395,9 @@ abstract class ColumnData<T> {
   /// Get the cell's tooltip value from the given [dataObject].
   String getTooltip(T dataObject) => getDisplayValue(dataObject);
 
+  /// Get the cell's text color from the given [dataObject].
+  Color getTextColor(T dataObject) => null;
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/table_data.dart:399]: not your fault but add a todo to the top of this class that says "remove get from method names"

**Comment by kenzieschmoll** [comment]
Also, can you add a layer of nesting to the fake data to make sure the "tree" aspect of both the treemap and table acts as expected?

**Comment by peterdjlee** [comment]
> Also, can you add a layer of nesting to the fake data to make sure the "tree" aspect of both the treemap and table acts as expected?

* Updated fake data with random children
Screen Shot 2020-07-01 at 3 38 58 PM

File: packages/devtools_app/lib/src/code_size/code_size_table.dart (hunk 5 above)
Comment (peterdjlee) [reply packages/devtools_app/lib/src/code_size/code_size_table.dart:135]: > How does this look in light mode?

* Light mode
Screen Shot 2020-07-01 at 3 39 04 PM

**Pull request merged**